### PR TITLE
[storage-if] Move db reader to storage/storage-interface

### PR DESCRIPTION
**Pull request by lightmark**
## Motivation

We're going to remove the dependency on libradb from most crates, by isolating the interface as traits into another crate.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

CI

### CODE REVIEW

File: storage/libradb/src/lib.rs [hunk 1]
```diff
@@ -807,6 +743,96 @@ impl LibraDB {
...
+        client_known_version: Version,
+        request_items: Vec<RequestItem>,
+    ) -> Result<(
+        Vec<ResponseItem>,
+        LedgerInfoWithSignatures,
+        ValidatorChangeProof,
+        AccumulatorConsistencyProof,
+    )> {
+        Self::update_to_latest_ledger(self, client_known_version, request_items)
+    }
```
Comment (msmouse) [inline storage/libradb/src/lib.rs]: possible to split these newly added interfaces to a separate commit / PR? (and how are they covered by tests?)

**Comment by msmouse** [comment]
So clean.

@bors-libra  r+

File: storage/storage-interface/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    account_address::AccountAddress,
+    account_state_blob::{AccountStateBlob, AccountStateWithProof},
+    contract_event::ContractEvent,
+    event::EventKey,
+    ledger_info::LedgerInfoWithSignatures,
+    proof::AccumulatorConsistencyProof,
+    transaction::{TransactionListWithProof, TransactionWithProof, Version},
+    validator_change::ValidatorChangeProof,
+};
+use storage_proto::StartupInfo;
```
Comment (davidiw) [inline storage/storage-interface/src/lib.rs:15]: can we make storage_proto depend on this?

**Comment by davidiw** [comment]
Nice PR, could we have a little more?
